go/cmd/dolt/commands/checkout.go

```diff
@@ -82,237 +83,66 @@ func (cmd CheckoutCmd) Exec(ctx context.Context, commandStr string, args []strin
-
-			return errhand.BuildDError(str, name).Build()
-		}
-		return errhand.BuildDError("error: could not find %s", name).Build()
-	} else if len(remoteRefs) == 1 {
-		verr := checkoutNewBranchFromStartPt(ctx, dEnv, name, remoteRefs[0].String())
-		if verr != nil {
-			return verr
-		}
-		headRef, err := dEnv.RepoStateReader().CWBHeadRef()
-		if err != nil {
-			return errhand.BuildDError(err.Error()).Build()
-		}
-		return SetRemoteUpstreamForBranchRef(dEnv, remoteRefs[0].GetRemote(), remoteRefs[0].GetBranch(), headRef)
-	} else {
-		// TODO : add hint of using `dolt checkout --track <remote>/<branch>` when --track flag is supported
-		return errhand.BuildDError("'%s' matched multiple (%v) remote tracking branches", name, len(remoteRefs)).Build()
-	}
-}
+	_, err = getRowsForSql(queryEngine, sqlCtx, sqlQuery)
 
-func checkoutNewBranchFromStartPt(ctx context.Context, dEnv *env.DoltEnv, newBranch, startPt string) errhand.VerboseError {
-	err := actions.CreateBranchWithStartPt(ctx, dEnv.DbData(), newBranch, startPt, false, nil)
-	if err != nil {
-		return errhand.BuildDError(err.Error()).Build()
+	var branchName string
+	if apr.NArg() > 0 {
+		branchName = apr.Arg(0)
 	}
-
-	return checkoutBranch(ctx, dEnv, newBranch, false)
+	return HandleVErrAndExitCode(handleErrors(branchName, err), usagePrt)
 }
 
-func checkoutTables(ctx context.Context, dEnv *env.DoltEnv, tables []string) errhand.VerboseError {
-	roots, err := dEnv.Roots(ctx)
-	if err != nil {
-		return errhand.VerboseErrorFromError(err)
-	}
+// generateCheckoutSql returns the query that will call the `DOLT_CHECKOUT` stored procedure.
+func generateCheckoutSql(args []string) (string, error) {
+	var buffer bytes.Buffer
+	queryValues := make([]interface{}, 0, len(args))
 
-	roots, err = actions.CheckoutTables(ctx, roots, tables)
-	if err != nil {
-		if doltdb.IsRootValUnreachable(err) {
-			return unreadableRootToVErr(err)
-		} else if actions.IsTblNotExist(err) {
-			badTbls := actions.GetTablesForError(err)
-			bdr := errhand.BuildDError("")
-			for _, tbl := range badTbls {
-				bdr.AddDetails("error: table '%s' did not match any table(s) known to dolt.", tbl)
-			}
-			return bdr.Build()
-		} else {
-			bdr := errhand.BuildDError("fatal: Unexpected error checking out tables")
-			bdr.AddCause(err)
-			return bdr.Build()
-		}
-	}
+	buffer.WriteString("CALL DOLT_CHECKOUT('--global'")
 
-	err = dEnv.UpdateWorkingRoot(ctx, roots.Working)
-	if err != nil {
-		return errhand.VerboseErrorFromError(err)
+	for _, arg := range args {
+		buffer.WriteString(", ?")
+		queryValues = append(queryValues, arg)
 	}
+	buffer.WriteString(")")
 
-	return nil
+	return dbr.InterpolateForDialect(buffer.String(), queryValues, dialect.MySQL)
 }
 
-func checkoutBranch(ctx context.Context, dEnv *env.DoltEnv, name string, force bool) errhand.VerboseError {
-	err := actions.CheckoutBranch(ctx, dEnv, name, force)
-
+func handleErrors(name string, err error) errhand.VerboseError {
```

Comment:
s/name/branchName/g

Also, enforce that an error is always passed in. Your verb/subject in the name says you are going to handle errors. It would never occur to me that there were no errors to handle based on those two words.

Reply:
Done.